guides/common/modules/proc_configuring-dynamic-dns-update-with-gss-tsig-authentication.adoc

```diff
@@ -138,24 +138,11 @@ grant {smart-proxy-principal}\047__{foreman-example-com}@EXAMPLE.COM__ wildcard
 .Configuring the {Project} or {SmartProxyServer} that manages the DNS service for the domain
 
 . Use the `{foreman-installer}` command to configure the {Project} or {SmartProxy} that manages the DNS Service for the domain:
-* On {Project}, enter the following command:
+* On {Project} or {SmartProxy}, enter the following command:
```

Comment:
I thought this would cover it:
```suggestion
* Configure your {ProjectServer} or {SmartProxyServer} to connect to your DNS service:
```

But then the header above it is:
> .Configuring the {Project} or {SmartProxyServer} that manages the DNS service for the domain

IMHO that's too similar.

Taking a step back, we have this situation:

Foreman knows about a Foreman Proxy. In this step we configure the Foreman Proxy DNS module. It does a few things:
* `--foreman-proxy-dns` enables the DNS module
* `--foreman-proxy-dns-provider` configures the DNS module to use the nsupdate_gss provider
* `--foreman-proxy-dns-managed` tells the installer not to install a DNS service
* `--foreman-proxy-dns-server` tells the DNS module where it can find the DNS server
* `--foreman-proxy-dns-tsig-keytab` & `--foreman-proxy-dns-tsig-principal` tells the DNS provider which credentials to use when connecting

I know that's too many details to cover, but I'm hoping it helps you to come up with a good suggestion.

Reply:
I think your suggestion would work just fine... if you merge `.Configuring the {Project} or {SmartProxyServer} [...]` with `.Updating the configuration in the {ProjectWebUI}`. Which is something I wanted to suggest before, but it seemed out of scope of this PR.

I couldn't find words to explain it, so I'm sending a patch (and I can only hope I'm doing this patch thing correctly :crossed_fingers:).

```patch
diff --git a/guides/common/modules/proc_configuring-dynamic-dns-update-with-gss-tsig-authentication.adoc b/guides/common/modules/proc_configuring-dynamic-dns-update-with-gss-tsig-authentication.adoc
index 94fdd8cbd9..3248c40880 100644
--- a/guides/common/modules/proc_configuring-dynamic-dns-update-with-gss-tsig-authentication.adoc
+++ b/guides/common/modules/proc_configuring-dynamic-dns-update-with-gss-tsig-authentication.adoc
@@ -137,8 +137,7 @@ grant {smart-proxy-principal}\047__{foreman-example-com}@EXAMPLE.COM__ wildcard
 
 .Configuring the {Project} or {SmartProxyServer} that manages the DNS service for the domain
 
-. Use the `{foreman-installer}` command to configure the {Project} or {SmartProxy} that manages the DNS Service for the domain:
-* On {Project} or {SmartProxy}, enter the following command:
+. Configure your {ProjectServer} or {SmartProxyServer} to connect to your DNS service:
 +
 [options="nowrap" subs="+quotes,attributes"]
 ----
@@ -150,17 +149,14 @@ grant {smart-proxy-principal}\047__{foreman-example-com}@EXAMPLE.COM__ wildcard
 --foreman-proxy-dns-tsig-principal="{smart-proxy-principal}/_{foreman-example-com}@EXAMPLE.COM_" \
 --foreman-proxy-dns=true
 ----
-
-After you run the `{foreman-installer}` command to make any changes to your {SmartProxy} configuration, you must update the configuration of each affected {SmartProxy} in the {ProjectWebUI}.
-
-.Updating the configuration in the {ProjectWebUI}
-. In the {ProjectWebUI}, navigate to *Infrastructure* > *{SmartProxies}*, locate the {ProductName}, and from the list in the *Actions* column, select *Refresh*.
-. Configure the domain:
-.. In the {ProjectWebUI}, navigate to *Infrastructure* > *Domains* and select the domain name.
-.. In the *Domain* tab, ensure *DNS {SmartProxy}* is set to the {SmartProxy} where the subnet is connected.
-. Configure the subnet:
-.. In the {ProjectWebUI}, navigate to *Infrastructure* > *Subnets* and select the subnet name.
-.. In the *Subnet* tab, set *IPAM* to *None*.
-.. In the *Domains* tab, select the domain that you want to manage using the IdM server.
-.. In the *{SmartProxies}* tab, ensure *Reverse DNS {SmartProxy}* is set to the {SmartProxy} where the subnet is connected.
-.. Click *Submit* to save the changes.
+. For each affected {SmartProxy}, update the configuration of that {SmartProxy} in the {ProjectWebUI}:
+.. In the {ProjectWebUI}, navigate to *Infrastructure* > *{SmartProxies}*, locate the {ProductName}, and from the list in the *Actions* column, select *Refresh*.
+.. Configure the domain:
+... In the {ProjectWebUI}, navigate to *Infrastructure* > *Domains* and select the domain name.
+... In the *Domain* tab, ensure *DNS {SmartProxy}* is set to the {SmartProxy} where the subnet is connected.
+.. Configure the subnet:
+... In the {ProjectWebUI}, navigate to *Infrastructure* > *Subnets* and select the subnet name.
+... In the *Subnet* tab, set *IPAM* to *None*.
+... In the *Domains* tab, select the domain that you want to manage using the IdM server.
+... In the *{SmartProxies}* tab, ensure *Reverse DNS {SmartProxy}* is set to the {SmartProxy} where the subnet is connected.
+... Click *Submit* to save the changes.
```

Now, as is usually the case, this creates a whole new problem, namely procedure steps nested in 3 levels. But I still think this is a cleaner solution, especially since I assume you never want users to just connect to the DNS service, but not update the config in the web UI right after.

WDYT?

Reply:
Yes, that's exactly the issue I had and I think your suggestion looks better.